lib/jsdom/living/nodes/HTMLProgressElement-impl.js

```diff
@@ -10,38 +10,58 @@ class HTMLProgressElementImpl extends HTMLElementImpl {
     this._labels = null;
   }
 
-  get value() {
-    const parsedValue = parseFloatingPointNumber(this.getAttributeNS(null, "value"));
+  get _isDeterminate() {
+    return this.hasAttributeNS(null, "value");
+  }
 
-    if (!isNaN(parsedValue) && parsedValue > 0) {
-      return parsedValue > this.max ? this.max : parsedValue;
+  // https://html.spec.whatwg.org/multipage/form-elements.html#concept-progress-value
+  get _value() {
+    const valueAttr = this.getAttributeNS(null, "value");
+    const parsedValue = parseFloatingPointNumber(valueAttr);
+    if (parsedValue !== null && parsedValue > 0) {
+      return parsedValue;
     }
+    return 0;
+  }
+
+  // https://html.spec.whatwg.org/multipage/form-elements.html#concept-progress-current-value
+  get _currentValue() {
+    const value = this._value;
+    return value > this.max ? this.max : value;
+  }
 
+  get value() {
+    if (this._isDeterminate) {
+      return this._currentValue;
+    }
     return 0;
   }
   set value(value) {
     this.setAttributeNS(null, "value", value);
   }
 
   get max() {
-    const parsedMax = parseFloatingPointNumber(this.getAttributeNS(null, "max"));
-
-    if (!isNaN(parsedMax) && parsedMax > 0) {
-      return parsedMax;
+    const max = this.getAttributeNS(null, "max");
+    if (max !== null) {
+      const parsedMax = parseFloatingPointNumber(max);
+      if (parsedMax !== null && parsedMax > 0) {
+        return parsedMax;
+      }
     }
-
     return 1.0;
   }
   set max(value) {
-    this.setAttributeNS(null, "max", value);
+    if (value > 0) {
+      this.setAttributeNS(null, "max", String(value));
```

Comment:
The string conversion is not necessary due to https://github.com/jsdom/jsdom/commit/4285e7074367c56dda07d3c08d6528ff0651cd4d#diff-802e42090f68edfaab3d0f57d32de76bR249

Reply:
Sure. I'd want to eventually remove the type cast though. Maybe after reflection gets re-done.

Reply:
I wouldn't be comfortable doing that unless we had some sort of static enforcement (e.g. lint rule) or auto-generated testing infrastructure to catch future errors... it's just too easy to forget that the useful behavior from the wrapper is omitted in the impl...